services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/trial.rb

```diff
@@ -110,9 +113,17 @@ def average_g_eval_score = scores.blank? ? 0 : (1.0 * scores.sum / scores.size).
         def scores = g_evals&.values&.flatten&.compact&.map(&:to_f)
 
         def update_results!(new_data)
```

Comment:
This section is what I am flagging for re-review. Doing the `save` then the `with_lock, reload, save` is the only way I've gotten it fully functioning, but am worried there might be side effects, so I'd appreciate another set of eyes.

Reply:
I think this is an improvement on current code, but I think long term we probably want to make some updates to potentially persisting or caching results.